src/app/dialogs/edit-poll-item-dialog/edit-poll-item-dialog.component.ts

```diff
@@ -0,0 +1,188 @@
+import {QuizItemCreate} from '../../model/poll-item-create/quiz-item-create';
+import {MultipleChoiceItemParticipant} from '../../model/poll-item-participant/multiple-choice-item-participant';
+import {OpenTextItemParticipant} from '../../model/poll-item-participant/open-text-item-participant';
+import {QuizItemParticipant} from '../../model/poll-item-participant/quiz-item-participant';
+
+// Constants
+const STEP_LABELS = [
+  'Update question / answers',
+  'Update options'
+];
+
+/**
+ * Wizard dialog, which lets the user edit an existing poll item.
+ * The dialog consists of two pages, whose appearance differ based on the selected item type.
+ */
+@Component({
+  selector: 'app-edit-poll-item-dialog',
+  templateUrl: './edit-poll-item-dialog.component.html',
+  styleUrls: ['./edit-poll-item-dialog.component.sass']
+})
+export class EditPollItemDialogComponent implements OnInit {
+
+  // Constant associations
+  stepLabels = STEP_LABELS;
+  optionTypes = OptionType;
+
+  // Event Emitters
+  @Input() isVisible: boolean;
+  @Input() poll: Poll;
+  @Input() pollItem: MultipleChoiceItemParticipant|OpenTextItemParticipant|QuizItemParticipant;
+  @Output() finish = new EventEmitter<boolean>(); // true = success; false = cancel
+
+  // Variables
+  step = 1;
+  loading = false;
+  errorMessage = '';
+  itemType = 0;
+  question = '';
+  answers = ['', ''];
+  options = [];
+
+  /**
+   * Initialize the component
+   *
+   * @param pollItemService Injected PollItemService
+   * @param tools Injected ToolsService
+   */
+  constructor(
+    private pollItemService: PollItemService,
+    private tools: CommonToolsService
+  ) {}
+
+  /**
+   * Initialize dialog
+   */
+  ngOnInit(): void {
+    if (this.pollItem instanceof MultipleChoiceItemParticipant) {
+      this.question = this.pollItem.question;
+      this.answers = this.pollItem.answers.map((answer) => answer.selectionOption);
+      this.itemType = 2;
+    }
+    if (this.pollItem instanceof QuizItemParticipant) {
+      this.question = this.pollItem.question;
+      this.answers = this.pollItem.answers.map((answer) => answer.selectionOption);
+      this.itemType = 3;
+    }
+    if (this.pollItem instanceof OpenTextItemParticipant) {
+      this.question = this.pollItem.question;
+      this.itemType = 1;
+    }
+  }
+
+  /**
+   * User clicked on 'Next' button.
+   * It handles the validity check of the entries on the current page and throws an
+   * error message or redirects the user to the next page if the user input is valid
+   */
+  handleNext(): void {
+    if (this.step === STEP_LABELS.length) {
+      this.step++;
+      this.editPollItem(this.poll.id, this.question, this.answers);
+    } else {
+      // Validity check
+      switch (this.step) {
+        case 1:
+          // Inputs are valid, set options data of selected item type
+          this.options = OPTIONS_DATA[this.itemType - 1].filter(option => option.visibleAtCreation);
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/livepoll/live-poll/pullRequest?prid=7591734)
```suggestion
          this.options = OPTIONS_DATA[this.itemType - 1].filter((option) => option.visibleAtCreation);
```